beetsplug/acousticbrainz.py

```diff
@@ -41,66 +46,118 @@ def func(lib, opts, args):
         cmd.func = func
         return [cmd]
 
+    def import_task_files(self, session, task):
+        """Function is called upon beet import.
+        """
+
+        items = task.imported_items()
+        fetch_info(self._log, items)
+
 
 def fetch_info(log, items):
-    """Currently outputs MBID and corresponding request status code
+    """Currently outputs MBID and corresponding request status code.
     """
+    def get_value(map_path):
+        try:
+            return reduce(lambda d, k: d[k], map_path, data)
+        except KeyError:
+            log.debug('Invalid Path: {}', map_path)
+
     for item in items:
         if item.mb_trackid:
             log.info('getting data for: {}', item)
 
             # Fetch the data from the AB API.
-            url = generate_url(item.mb_trackid)
-            log.debug('fetching URL: {}', url)
+            urls = [generate_url(item.mb_trackid, path) for path in LEVELS]
+            log.debug('fetching URLs: {}', urls)
             try:
-                rs = requests.get(url)
+                responses = [requests.get(url) for url in urls]
             except requests.RequestException as exc:
                 log.info('request error: {}', exc)
                 continue
 
             # Check for missing tracks.
-            if rs.status_code == 404:
+            if any(r.status_code == 404 for r in responses):
                 log.info('recording ID {} not found', item.mb_trackid)
                 continue
 
             # Parse the JSON response.
             try:
-                data = rs.json()
+                data = responses[0].json()
+                data.update(responses[1].json())
             except ValueError:
-                log.debug('Invalid Response: {}', rs.text)
+                log.debug('Invalid Response: {} & {}',
+                          responses[0].text, responses[1].text)
```

Comment:
And this could be generalized as: `log.debug('Invalid Responses: {}', [r.text for r in responses])`